plugins/pixart-rf/fu-pxi-device.c

```diff
@@ -126,7 +149,32 @@ fu_pxi_device_calculate_checksum (const guint8 *data, gsize sz)
 	return checksum;
 }
 
-/* FIXME: is @port required? */
+static gboolean
+fu_pxi_device_check_support_resume (FuPxiDevice *self,
+				 	FuFirmware *firmware,
+				  	GError **error)
+{
+	g_autoptr(GBytes) fw = NULL;
+	g_autoptr(GPtrArray) chunks = NULL;
+	guint16 checksum_tmp = 0x0;
+	
+	/* get the default image */
+	fw = fu_firmware_get_image_default_bytes (firmware, error);
+	chunks = fu_chunk_array_new_from_bytes (fw, 0x0, 0x0, FU_PXI_DEVICE_OBJECT_SIZE_MAX);
+	
+	/* calculate device current checksum */
+	for (guint i = 0; i < self->offset; i++) {	
```

Comment:
I think you want `i < chunks->len`, no?

Reply:
no the self-offset is from device , i could be zero or other value.
some time BLE could disconnect we can check last checksum in device , and continue to transmit.

Reply:
It could transmit 2 object and then disconnect suddenly.When the next update we get the offset is 2, it will start offset 2 if the file is same with last time

Reply:
Ahh, cool! In which case you need to check if the offset > chunks->len before you start in case the new firmware is shorter and you over-read the buffer.